src/hooks/useSearchManager.js

```diff
@@ -1,155 +1,150 @@
-import { useContext } from 'react'
+import { useContext, useMemo } from 'react'
 import { useRouter } from 'next/router'
 import { SearchManagerContext } from 'contexts/SearchManagerContext'
 import { options } from 'config'
+import parseDefaultSearchParams from 'helpers/parseDefaultSearchParams'
 
 export const useSearchManager = () => {
+  const { facetNames, setFacetNames, searchParams, setSearchParams } =
+    useContext(SearchManagerContext)
   const {
-    facetNames,
-    setFacetNames,
-    search: searchState,
-    setSearch: setSearchState,
-    filterOrders: filterOrdersState,
-    setFilterOrders: setFilterOrdersState
-  } = useContext(SearchManagerContext)
-  const {
-    search: { defaultOrdering, hasPublication, numDownloadableSamples }
+    search: { numDownloadableSamples }
   } = options
   const router = useRouter()
-  const filterOrders = filterOrdersState
-  const setFilterOrders = setFilterOrdersState
-  const search = searchState
-  const setSearch = setSearchState
 
   /* Common */
-  const resetPage = () => {
-    search.offset = 0
+  const updatePage = (page) => {
+    const offset = (page - 1) * searchParams.limit
+    if (offset === searchParams.offset) return
 
-    setSearch({ ...search })
+    setSearchParams({
+      ...searchParams,
+      offset
+    })
   }
 
-  const updatePage = (newPage) => {
-    search.offset = (newPage - 1) * search.limit
+  // resets page on page size changes
+  const updatePageSize = (limit) => {
+    if (limit === searchParams.limit) return
 
-    setSearch({ ...search })
-    updateSearchQuery()
+    setSearchParams({
+      ...searchParams,
+      limit,
+      offset: 0
+    })
   }
 
-  const updatePageSize = (newPageSize) => {
-    search.limit = newPageSize
+  const updateSortBy = (ordering) => {
+    if (ordering === searchParams.ordering) return
 
-    setSearch({ ...search })
-    updateSearchQuery(true)
+    setSearchParams({
+      ...searchParams,
+      ordering
+    })
   }
 
-  const updateSortBy = (newSortOrder) => {
-    if (newSortOrder === defaultOrdering) {
-      delete search.ordering
-    } else {
-      search.ordering = newSortOrder
-    }
-
-    setSearch({ ...search })
-    updateSearchQuery()
+  // update a single parameter value
+  const updateParamValue = (paramName, newValue) => {
+    setSearchParams({
+      ...searchParams,
+      [paramName]: newValue
+    })
```

Comment:
I've defined a separate method here for the `num_downloadable_samples__gt` key, as it's no longer part of the facet filter items, and removed its filtering logic from `toggleFilter` accordingly. The method is generic and simply updates the value of a single search parameter to the given value. 

Let me know if you have any suggestions for a better name for this method!

Reply:
`updateSearchParam` would work, since we are updating an arbitrary param.